pymemcache/client/base.py

```diff
@@ -236,6 +241,10 @@ def __init__(self,
                  no_delay=False,
                  ignore_exc=False,
                  socket_module=socket,
+                 socket_keepalive=False,
+                 socket_keepaliveidle=1,
+                 socket_keepaliveintvl=1,
+                 socket_keepalivecnt=5,
```

Comment:
Would these read better with an extra underscore?

```suggestion
                 socket_keepalive_idle=1,
                 socket_keepalive_intvl=1,
                 socket_keepalive_cnt=5,
```

Alternatively, because these are only relevant for keepalive, maybe we should define a single configuration structure and use that here instead.

```py
class KeepaliveOpts(object):
    __slots__ = ('idle', 'intvl', 'cnt')

    def __init__(self, idle=1, intvl=1, cnt=5):
        self.idle = idle
        self.intvl = intvl
        self.cnt = cnt
```

Or simply a dictionary of options:

```py
# Defaults:
SOCKET_KEEPALIVE_DEFAULTS = {
    socket.TCP_KEEPIDLE: 1,
    socket.TCP_KEEPINTVL: 1,
    socket.TCL_KEEPCNT: 5,
}

# Override based on provided argument (socket_keepalive_opts):

self.socket_keepalive_opts = SOCKET_KEEPALIVE_DEFAULTS
self.socket_keepalive_opts.update(socket_keepalive_opts)
```

What do you think?